qiime2/sdk/usage.py

```diff
@@ -8,60 +8,194 @@
 
 import abc
 import re
-import types
+from types import MappingProxyType
+from typing import Tuple, Callable, Optional, Union, List, Set, Dict
```

Comment:
Stylistically, I prefer `import types`, then using `types.MappingProxyType` - I like this approach (esp with external deps) because it shows where the object comes from, inline.

Reply:
Fair enough!  Do you prefer this for both `types` and `typing`?

Reply:
Yes (the comment was on both lines, although it isn't very clear here - maybe GH only formats that specially when suggesting a code change?)

Reply:
I saw that the comment was on both lines but just wanted to make sure :)